cairo-oracle-hint-processor/src/rpc_1_hint_processor/mod.rs

```diff
@@ -96,11 +98,35 @@ impl<'a> Rpc1HintProcessor<'a> {
             .post(self.server.as_ref().unwrap())
             .json(&data)
             .send()
+            .map_err(|_| {
+                format!(
+                    "Error sending request to oracle server {}",
+                    self.server.as_ref().unwrap()
+                )
+            })
             .unwrap()
             .json::<Value>()
+            .map_err(|_| {
+                format!(
+                    "Error reading response from oracle server {}",
+                    self.server.as_ref().unwrap()
+                )
+            })
             .unwrap();
 
-        let output = &resp["result"];
+        let resp = resp
+            .as_object()
+            .expect("Error serialising response as object from oracle server.");
+
+        if resp.keys().count() != 1 {
+            panic!(
+                "Response object from oracle server shall contain only the key `result`. More than one key found."
+            )
+        }
+
+        let output = resp
+            .get("result")
+            .expect("Key `result` not found in response from oracle server.");

```

Comment:
Print what we've received instead. For example, it's totally reasonable for the RPC server to return something like:

```{ "error": "validation error, number cannot be negative" }```

We want users to be able to read their own error messages.